modules/Students/student_view_details.php

```diff
@@ -323,11 +323,9 @@
 						$subpage="Overview" ;
 					}
 					
-					if ($search!="") {
-						print "<div class='linkTop'>" ;
-							print "<a href='" . $_SESSION[$guid]["absoluteURL"] . "/index.php?q=/modules/Students/student_view.php&search=" . $search . "'>" . _('Back to Search Results') . "</a>" ;
-						print "</div>" ;
-					}
+					print "<div class='linkTop'>" ;
+						print "<a href='" . $_SESSION[$guid]["absoluteURL"] . "/index.php?q=/modules/Students/student_view.php&search=$search&allStudents=$allStudents&sort=$sort&subpage=$subpage&category=" . "'>" . _('Back to Search Results') . "</a>" ;
```

Comment:
This will now show even if search is not set, which we don't want. Can you leave the if, but check to see if search or sort are blank, so it only shows when one isn't?